grpo训练中eval_use_evalscope不生效

### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

训练脚本中添加了参数：
--eval_use_evalscope \
--eval_dataset "amc" "aime_24" "math_500" \
--eval_generation_config "{"do_sample":true,"temperature":1,"top_p":0.7}" \

但是出现以下错误：
[rank3]:   File "/mnt/workspace/wuhuacong/ms-swift/swift/rlhf_trainers/grpo_trainer.py", line 119, in __init__
[rank3]:     assert len(self.eval_dataset) >= total_eval_batch_size, (
[rank3]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[rank3]: AssertionError: eval_dataset size 0 is smaller than total_eval_batch_size 8. Please increase the size of eval_dataset or set a larger value for split_dataset_ratio.

### How to Reproduce / 如何复现

swift rlhf \
    --rlhf_type "grpo" \
    --model "Qwen2.5-Math-7B" \
    --tuner_type "full" \
    --dataset "competition_math/data/train-00000-of-00001-7320a6f3aba8ebd2.parquet" \
    --use_vllm "true" \
    --vllm_mode "colocate" \
    --vllm_gpu_memory_utilization "0.5" \
    --vllm_tensor_parallel_size "4" \
    --vllm_enforce_eager "true" \
    --torch_dtype "bfloat16" \
    --num_train_epochs "10" \
    --per_device_train_batch_size "4" \
    --per_device_eval_batch_size "4" \
    --gradient_accumulation_steps "16" \
    --generation_batch_size "1024" \
    --learning_rate "5e-6" \
    --save_steps "20" \
    --save_total_limit "4" \
    --logging_steps "1" \
    --warmup_ratio "0.05" \
    --lr_scheduler_type "cosine" \
    --dataloader_num_workers "20" \
    --max_completion_length "4096" \
    --resume_only_model "false" \
    --save_only_model "false" \
    --reward_funcs "accuracy" \
    --reward_weights "1" \
    --num_generations "16" \
    --num_generations_eval "4" \
    --sleep_level "1" \
    --temperature "1.0" \
    --top_p "0.85" \
    --greater_is_better "true" \
    --metric_for_best_model "eval_reward" \
    --deepspeed "zero1" \
    --report_to "wandb" \
    --eval_use_evalscope \
    --eval_dataset "amc" "aime_24" "math_500" \
    --eval_generation_config "{"do_sample":true,"temperature":1,"top_p":0.7}" \

### Additional Information / 补充信息

_No response_

--- a/swift/rlhf_trainers/grpo_trainer.py
+++ b/swift/rlhf_trainers/grpo_trainer.py
@@ -112,7 +112,7 @@ class GRPOTrainer(RolloutTrainerMixin, SwiftMixin, HFGRPOTrainer):
         if not self.reward_funcs and not self.use_gym_env:
             raise ValueError('You must specify reward_funcs or reward_model')
 
-        if self.args.eval_strategy != 'no':
+        if self.args.eval_strategy != 'no' and not self.args.eval_use_evalscope:
             total_eval_batch_size = self.args.per_device_eval_batch_size * \
                 self.accelerator.num_processes // self.num_generations_eval
             assert len(self.eval_dataset) >= total_eval_batch_size, (

